libs/librepcb/projecteditor/schematiceditor/schematiceditor.cpp

```diff
@@ -696,31 +696,51 @@ void SchematicEditor::renameSchematic(int index) noexcept {
   }
 }
 
-QStringList SchematicEditor::getSearchToolBarCompleterList() noexcept {
-  QStringList list;
+QList<SI_Symbol*> SchematicEditor::getSearchCandidates() noexcept {
+  QList<SI_Symbol*> candidates = {};
   foreach (const Schematic* schematic, mProject.getSchematics()) {
-    foreach (const SI_Symbol* symbol, schematic->getSymbols()) {
-      list.append(symbol->getName());
+    Q_ASSERT(schematic);
+    foreach (SI_Symbol* symbol, schematic->getSymbols()) {
+      candidates.append(symbol);
     }
   }
+  std::sort(candidates.begin(), candidates.end(),
+            [](SI_Symbol* a, SI_Symbol* b){
+    return a->getName() < b->getName();
+  });
+  return candidates;
+}
+
+QStringList SchematicEditor::getSearchToolBarCompleterList() noexcept {
+  QStringList list;
+  foreach (SI_Symbol* symbol, getSearchCandidates()) {
+    list.append(symbol->getName());
+  }
   return list;
 }
 
-void SchematicEditor::goToSymbol(const QString& name) noexcept {
-  for (int i = 0; i < mProject.getSchematics().count(); ++i) {
-    const Schematic* schematic = mProject.getSchematicByIndex(i);
-    Q_ASSERT(schematic);
-    foreach (SI_Symbol* symbol, schematic->getSymbols()) {
-      if (symbol->getName().toLower() == name.toLower()) {
-        setActiveSchematicIndex(i);
-        schematic->clearSelection();
-        symbol->setSelected(true);
-        QRectF rect   = symbol->getBoundingRect();
-        qreal  margin = Length(1000000).toPx();
-        rect.adjust(-margin, -margin, margin, margin);
-        mGraphicsView->zoomToRect(rect);
-        return;
-      }
+void SchematicEditor::goToSymbol(const QString& name,
+                                 unsigned int index) noexcept {
+  QList<SI_Symbol*> symbolCandidates = {};
+  foreach (SI_Symbol* symbol, getSearchCandidates()) {
+    if (symbol->getName().startsWith(name, Qt::CaseInsensitive)) {
+      symbolCandidates.append(symbol);
+    }
+  }
+
+  if (symbolCandidates.count()) {
+    index %= symbolCandidates.count();
+    SI_Symbol* symbol = symbolCandidates[index];
+    Schematic* schematic = &symbol->getSchematic();
+    if (setActiveSchematicIndex(mProject.getSchematics().indexOf(schematic))) {
+      schematic->clearSelection();
```

Comment:
As mentioned in a previous PR, if you get a reference, better don't convert it to a pointer if not really needed:

```suggestion
    Schematic& schematic = symbol->getSchematic();
    if (setActiveSchematicIndex(mProject.getSchematics().indexOf(&schematic))) {
      schematic.clearSelection();
```

Of course it's not critical, but IMHO it's good habit to let references stay references :)